src/engine/new.go

```diff
@@ -0,0 +1,53 @@
+package engine
+
+import (
+	"oh-my-posh/color"
+	"oh-my-posh/console"
+	"oh-my-posh/platform"
+	"oh-my-posh/shell"
+)
+
+// NewEngine returns a prompt engine initialized with the
+// given configuration options, and is ready to print any
+// of the prompt components.
+func NewEngine(config string) *Engine {
+	env := &platform.Shell{
+		CmdFlags: &platform.Flags{
+			Config: config,
+			Shell:  "shell",
```

Comment:
this needs to be exposed imho

Reply:
Usually package writers expose functionality through some `pkg` package directory.
I have no particular suggestion as to where this should be exposed, you probably are better placed to decide.